### PR TITLE
Auth documentation

### LINKED ISSUE #329: Authentication
We have to implement authentication for PermaplanT.

This involves multiple steps:

- [x] frontend (#320)
- [x] #336 
  - [x] fix backend CORS configuration (is only necessary for local dev)
- [x] Keycloak (no code changes just deployed)
- [ ] add/refactor documentation #412 
- [ ] describe how to run benchmarks (@kitzbergerg) #392 

We have to set up all of them in one step otherwise they won't work together (e.g. backend requires auth but frontend hasn't yet implemented.

@markus2330 I would start with a setup in the backend that works locally. I'll leave it disabled and then once frontend oauth2 is ready we just have to enable it in that PR.

### LINKED ISSUE #392: documentation how to obtain tokens for local development
e.g. OAuth2 via Postman or curl

### PR DESCRIPTION
**Pull request by kitzbergerg**
Documentation for authentication in the backend.
The documentation is supposed to be viewed with mdbook. Navigation is set up that way.
Resolves #329, resolves #392.

## Basics

- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.
- [x] The PR is rebased with current master.

## Checklist

- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by kitzbergerg** [comment]
jenkins build please

**Review by lukashartl — APPROVED** [review]
lgtm

File: doc/backend/02obtain_access_tokens.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+A more in depth explanation about Postman and `OAuth 2.0` can be found in the [Postman documentation](https://learning.postman.com/docs/sending-requests/authorization/oauth-20/).
+
+## Using Curl
+
+The following request should work without problems:  
+`curl 'http://localhost:8080/api/config'`
+
+The following request should fail:  
+`curl 'http://localhost:8080/api/seeds'`
```
Comment (markus2330) [inline doc/backend/02obtain_access_tokens.md:36]: Does this actually fail for you? I simply don't get any response but `curl` executes successfully. (Tried on master).

File: doc/backend/02obtain_access_tokens.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  --data 'client_id=localhost'
+```
+
+The response should be JSON containing a key `access_token`.  
+Copy the access token.  
+You can now make the request like the following:  
+`curl 'http://localhost:8080/api/seeds' -H "authorization: Bearer {access_token}"`
+
+Note that the token is only valid for 5 minutes after which you have to refresh it.
+
```
Comment (markus2330) [inline doc/backend/02obtain_access_tokens.md:57]:
```suggestion
Note that the token is only valid for 5 minutes after which you have to refresh it with the same curl command again.

```

File: doc/backend/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# PermaplanT Backend
+
+## Requirements
+
+- Rust 1.67.1 or later ([Installation guide](../development_setup.md))
+- [PostgreSQL](https://www.postgresql.org/download/) version 13 or later
```
Comment (markus2330) [inline doc/backend/README.md:6]:
```suggestion
- [PostgreSQL](https://www.postgresql.org/download/) version 13 with PostGIS 3.1.1

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great job, only one clarification needed! Usecase needs to be updated, as we now don't enter "Nextcloud login credentials". permalinks will not exist but we will have public maps.

File: doc/backend/03api_documentation.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+- Click `Authorize`.
+
+- Use the variant with header `oauth2 (OAuth2, authorizationCode)`.
+- Enter the `client_id`. You have to use a different one depending on the location:
+  - Production: `PermaplanT-Prod`.
+  - PR: `PermaplanT-PR`.
+  - Dev: `PermaplanT-Dev`.
+  - Local:
+    - `local-swagger-ui` if you use the preconfigured Keycloak instance.
+    - If you configured Keycloak yourself you have to create a client matching the URL.
```
Comment (kitzbergerg) [inline doc/backend/03api_documentation.md:27]: @lukashartl Please check if the client_ids are correct as described here.

Comment (lukashartl) [reply doc/backend/03api_documentation.md:27]: They are correct

**Pull request merged**